src/functions.rs

```diff
@@ -1139,18 +1161,33 @@ fn call_to_resource_transition<'a>(
         .next()
         .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
         .get_list(context)?;
+    let obj_name = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?;
 
     if args_iter.next().is_some() {
         return Err(ErrorItem::Internal(InternalError::new()).into());
     }
 
     for file_type in file_types {
-        ret.push(ResourcetransRule {
-            default: Cow::Owned(default.clone()),
-            domain: Cow::Owned(domain.clone()),
-            parent: Cow::Owned(parent.clone()),
-            file_type: Cow::Owned(file_type.clone()),
-        });
+        if obj_name.to_string().is_empty() {
```

Comment:
Rather than copy/pasting, let's just do something like:

```
let obj_name = if obj_name.is_empty() {
    None
} else {
    Some(obj_name)
}
```